src/Parsers/ParserDropFunctionQuery.cpp

```diff
@@ -36,6 +36,7 @@ bool ParserDropFunctionQuery::parseImpl(IParser::Pos & pos, ASTPtr & node, Expec
     {
         if (!ASTQueryWithOnCluster::parse(pos, cluster_str, expected))
             return false;
+        pos.get();
```

Comment:
I don't understand this change.

Reply:
`ASTQueryWithOnCluster::parse` read identifier from ast(`DB::parseIdentifierOrStringLiteral`), leave `last_accessed_index` in `pos` not change. This cause `DB::tryParseQuery`(`const auto last_token = token_iterator.max();`) get the ending semicolon as new statement.
This access change the `last_accessed_index` value, so the ending semicolon will not be treat as new statement.
If `ASTQueryWithOnCluster::parse` is followed by other `ParserKeyword` parser, there will be no problem. `ParserCreateFunctionQuery` and other cluster parse is this situation.

Reply:
Sounds non-obvious. Is there a way to write it more explicitly?

Reply:
you mean add some comments on code? or I should add some detail for previous explain?

Reply:
I wrote more detail info about this on issue #45303 , please check.

The missing release binary cause merge check fail, as follow pic show:
![image](https://github.com/ClickHouse/ClickHouse/assets/1409809/711a072b-0760-4f84-aa45-2c24cfae7ab3)
![image](https://github.com/ClickHouse/ClickHouse/assets/1409809/59abe902-abd6-40f4-ae17-7e73d52d4a67)
Please advice me how to fix this.

@alexey-milovidov